tools/topology/topology2/doc/README

```diff
@@ -0,0 +1,7 @@
+To build the topology2 source documentation do following steps:
+
+cd tools/topology/topology2/doc
+cmake -B build
+cmake --build build
```

Comment:
Did you ever try these commands? There's a typo in the file name!

```
CMake Error: The source directory 
"sof/tools/topology/topology2/doc" does not appear to contain CMakeLists.txt.
```

Reply:
I've run "rm -rf build ; cmake -B build ; cmake --build build" in sof/tools/topology/topology2/doc multiple times last night when testing my changes.

But now I see what my problem, also with cmake rules is. I've been all the time editing CMakeList.txt not CMakeLists.txt... editing it again and retrying, never seeing any change, and finally committing it. While all the time CMakeLists.txt is a left over from some earlier attempt.

I'll try to pull this together and make another attempt at generating contents.doxy with cmake, this time editing the correct file.